compiler/src/compilers/compiler.rs

```diff
@@ -0,0 +1,7 @@
+use super::super::data::input_data::InputData;
+use super::super::data::output_data::OutputData;
+
+
+pub trait Compiler {
```

Comment:
Is it possible to use `Template` pattern to implement compiler?

Reply:
Actually, trait Compiler is the attempt to use Template in Handler module:

https://github.com/ITA-Dnipro/CompilerWebApp/blob/e8b7cf7480209a77980003fa75a2d0de2418625e/compiler/src/handler/mod.rs#L76-L86

In Compiler Explorer there are also some kind of the "Parent" compiler - BaseCompiler 

https://github.com/compiler-explorer/compiler-explorer/blob/84fa2fe598d94733a281fcf9410912daa1d3de82/lib/base-compiler.js#L49

Our compiler uses another approach, but the same idea. 
Will we get any critical issues with such Template?